lib/ansible/modules/storage/netapp/sf_volume_manager.py

```diff
@@ -0,0 +1,395 @@
+                changed = True
+
+            elif self.state == 'present':
+                if volume_detail.access is not None and self.access is not None and volume_detail.access != self.access:
+                    logger.debug("CHANGED: Volume access needs to be updated")
+                    update_volume = True
+                    changed = True
+
+                elif volume_detail.account_id is not None and self.new_account_id is not None \
+                        and volume_detail.account_id != self.new_account_id:
+                    logger.debug("CHANGED: Account ID for volume needs to be updated")
+                    update_volume = True
+                    changed = True
+
+                elif volume_detail.qos is not None and self.qos is not None and volume_detail.qos != self.qos:
+                    logger.debug("CHANGED: Volume QOS needs to be updated")
+                    update_volume = True
+                    changed = True
+
+                elif volume_detail.total_size is not None and volume_detail.total_size != self.size:
+                    size_difference = abs(float(volume_detail.total_size - self.size))
+                    # Change size only if difference is bigger than 0.001
+                    if size_difference/self.size > 0.001:
+                        log_msg = "CHANGED: Volume needs to be re-sized. Reported size is %s and new size is %s" % \
+                                  (str(volume_detail.total_size), str(self.size))
+
+                        logger.debug(log_msg)
+                        update_volume = True
+                        changed = True
+
+                elif volume_detail.attributes is not None and self.attributes is not None and \
+                        volume_detail.attributes != self.attributes:
+                    logger.debug("CHANGED: Volume attributes need to be updated")
+                    update_volume = True
+                    changed = True
+
+        else:
+            if self.state == 'present':
+                logger.debug(
+                    "CHANGED: volume does not exist, but requested state is "
+                    "'present'")
+
+                changed = True
+
+        if changed:
+            if self.module.check_mode:
+                logger.debug('skipping changes due to check mode')
+            else:
+                if self.state == 'present':
+                    if not volume_exists:
+                        self.create_volume()
+                    elif update_volume:
+                            self.update_volume()
+
+                elif self.state == 'absent':
+                    self.delete_volume()
+        else:
+            logger.debug("exiting with no changes")
+
+        self.module.exit_json(changed=changed)
+
+
+def main():
+    v = SolidFireVolume()
+
+    try:
+        v.apply()
+    except:
+        err = get_exception()
+        logger.debug("Exception in apply(): \n%s" % format_exc(err))
+        raise
```

Comment:
a module should never end in exception, it should use fail_json instead (you can pass traceback info if needed in 'exception' key)